src/main/java/seedu/contax/ui/onboarding/OnboardingWindow.java

```diff
@@ -119,126 +117,270 @@ public void translate(double x, double y) {
                     processStep(s);
-                    step.setEventType(s.getPositionOption());
+                    processInstructionPosition(s.getPositionOption());
                 }
             }
-            storyManager.stepFront();
-        }
-
+        });
     }
 
     /**
-     * Highlight the stage based on the given option
+     * Highlights a node in this window the stage based on the given option
+     * <br><br>
+     * Options are as follows:
+     * <br>- 0: clear all highlights
+     * <br>- 1: highlight only the command box
+     * <br>- 2: highlight only the person list
      * @param option Highlight option
      */
-    public void processHighlightOption(int option) {
-        if (option == 0) {
+    private void processHighlightOption(int option) {
+        switch (option) {
+        case 0:
             commandBox.unhighlight();
             commandBox.clear();
             personList.setStyle("-fx-border-width: 0px");
-        } else if (option == 1) {
+            break;
+        case 1:
             commandBox.highlight();
-        } else if (option == 2) {
+            break;
+        case 2:
             commandBox.unhighlight();
             commandBox.clear();
             personList.setStyle("-fx-border-color: yellow; -fx-border-width: 5px");
+            break;
+        default:
+            break;
         }
     }
 
     /**
-     * Overlay the stage based on the given option
-     * @param option Overlay option
+     * Displays an overlay on this window, whose position is dependant on the given option
+     * <br><br>
+     * Options are as follows:
+     * <br>- 0: cover all
+     * <br>- 1: cover all, showing only the menu bar
+     * <br>- 2: cover all, showing only the command box
+     * <br>- 3: cover all, showing only the person list
+     * @param option the overlay option
      */
-    public void processOverlayOption(int option) {
-        if (option == 0) {
+    private void processOverlayOption(int option) {
+        switch (option) {
+        case 0:
             coverAll();
-        } else if (option == 1) {
-            showOnly(menuBar);
-        } else if (option == 2) {
-            showOnly(commandBoxPlaceholder);
-        } else if (option == 3) {
-            showOnly(personList);
+            break;
+        case 1:
+            showOnly(menuBar, 0);
+            break;
+        case 2:
+            showOnly(commandBoxPlaceholder, 0);
+            break;
+        case 3:
+            showOnly(personList, 2);
+            break;
+        default:
+            break;
         }
     }
 
     /**
-     * Moves the InstructionLabel based on the given option
+     * Moves the InstructionLabel to a position based on the given option.
+     * <br><br>
+     * Options are as follows:
+     * <br>- 0: center of window
+     * <br>- 1: top right of menu bar
+     * <br>- 2: top right of command box
+     * <br>- 3: top right of result display
+     * <br>- 4: middle right of result display
```

Comment:
Would be better if you abstract it into an Enum, which can be nested in OnboardingStory if necessary.

Reply:
Great suggestion. Updated accordingly